### PR TITLE
Adding S3 integration

### PR DESCRIPTION
**Pull request by mdbenjam**
Using added S3 integration in caseflow-commons to provide a level of caching. First look at https://github.com/department-of-veterans-affairs/caseflow-commons/pull/17

**Test Plan**
- [x] Write new tests
- [x] Run test suite
- [ ] Help with integration on prod

### CODE REVIEW

File: app/models/document.rb [hunk 1]
```diff
@@ -38,19 +38,30 @@ def self.from_vbms_document(vbms_document)
...
   def content
-    @content ||= Appeal.repository.fetch_document_file(self)
+    return @content unless @content.nil?
+
+    @content = S3Service.fetch_content(file_name)
+    return @content unless @content.nil?
+
+    @content = Appeal.repository.fetch_document_file(self)
+    S3Service.store_file(file_name, @content)
+    @content
```
Comment (shanear) [inline app/models/document.rb]: I'd probably break this up a little. Something like this:

```
def fetch_and_cache_document_from_vbms(filename)
  Appeal.repository.fetch_document_file(self)
  S3Service.store_file(file_name, @content)
end

def fetch_content(file_name)
   S3Service.fetch_content(file_name) || fetch_and_cache_document_from_vbms(filename)
end

def content 
  @content ||= fetch_content(file_name)
end
```

File: spec/models/document_spec.rb [hunk 2]
```diff
@@ -60,28 +65,39 @@
...
+  context "#file_name" do
+    it "returns correct path" do
+      expect(document.file_name).to match(/123/)
+    end
+  end
 
+  context "#default_path" do
     it "returns correct path" do
-      expect(document.default_path).to match(/.*nod-123.pdf/)
+      expect(document.default_path).to match(/.*123/)
```
Comment (shanear) [inline spec/models/document_spec.rb]: this doesn't seem to be testing the logic of this function, which is hardcoding a filepath. I'd make the test a little more explicit. Something like this?

```
expect(document.default_path).to match(/.*\/tmp\/pdfs\/.*/123/)
```

File: app/models/document.rb (hunk 1 above)
Comment (shanear) [inline app/models/document.rb]: This will have the added bonus of making your tests a little less nested.

File: config/initializers/s3.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+require 'caseflow'
+
+S3Service = (Rails.application.config.s3_enabled ? Caseflow::S3Service : Caseflow::Fakes::S3Service)
```
Comment (orischwartz) [inline config/initializers/s3.rb]: @mdbenjam is it possible this file is misnamed and that's what's causing the `LoadError: Unable to autoload constant S3Service` error?

Comment (mdbenjam) [inline config/initializers/s3.rb]: Yeah, I think if we point to the latest version of commons things should clear up. https://github.com/department-of-veterans-affairs/caseflow/pull/571

**Pull request merged**